src/Runtime/LambdaRuntime.php

```diff
@@ -0,0 +1,154 @@
+<?php declare(strict_types=1);
+
+namespace Bref\Runtime;
+
+/**
+ * Client for the AWS Lambda runtime API.
+ *
+ * This allows to interact with the API and:
+ *
+ * - fetch events to process
+ * - signal errors
+ * - send invocation responses
+ *
+ * It is intentionally dependency-free to keep cold starts as low as possible.
+ *
+ * Usage example:
+ *
+ *     $lambdaRuntime = LambdaRuntime::fromEnvironmentVariable();
+ *     $lambdaRuntime->processNextEvent(function ($event) {
+ *         return <response>;
+ *     });
+ */
+class LambdaRuntime
+{
+    /** @var string */
+    private $apiUrl;
+
+    public static function fromEnvironmentVariable(): self
+    {
+        return new self(getenv('AWS_LAMBDA_RUNTIME_API'));
+    }
+
+    public function __construct(string $apiUrl)
+    {
+        if ($apiUrl === '') {
+            die('At the moment lambdas can only be executed in an Lambda environment');
+        }
+
+        $this->apiUrl = $apiUrl;
+    }
+
+    public function processNextEvent(callable $handler): void
+    {
+        [$invocationId, $event] = $this->waitNextInvocation();
+
+        try {
+            $this->sendResponse($invocationId, $handler($event));
+        } catch (\Throwable $e) {
+            $this->signalFailure($invocationId, $e);
+        }
+    }
+
+    /**
+     * Wait for the next lambda invocation and retrieve its data.
+     *
+     * This call is blocking because the Lambda runtime API is blocking.
+     *
+     * @see https://docs.aws.amazon.com/lambda/latest/dg/runtimes-api.html#runtimes-api-next
+     */
+    private function waitNextInvocation(): array
+    {
+        $handler = curl_init("http://{$this->apiUrl}/2018-06-01/runtime/invocation/next");
+        curl_setopt($handler, CURLOPT_FOLLOWLOCATION, true);
```

Comment:
would it make sense to use http2 for this requests?

Reply:
HTTP2 for the lambda runtime API? Is it working or documented somewhere?

Reply:
I asked for feedback on the docs and hope they will clarify what is supported (https and http2)

Reply:
Honestly I don't see any reason they will want to support HTTPS (or HTTP2) in this context since this is an internal (low-level) API.

Reply:
Let's mark this resolved for now and reopen if there's any reason to.